### PR TITLE
Plant list

### PR DESCRIPTION
**Pull request by FabianAlleblas**
I added a PlantList page with the components PlantCard and Header.
For the PlantCard: Until I've a better solution to keep the proportions of the images I stick with them as a background and fixed pixels.

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -1,5 +1,16 @@
+import { Route, Switch } from 'react-router-dom'
+import PlantListPage from './plantlistpage/PlantListPage'
+
 function App() {
-  return <div className="App"></div>
+  return (
+    <div>
```
Comment (felix-horn) [inline frontend/src/App.js:6]: What do you need the div for? Isn't it possible to wrapp the whole thing wiht just the Switch element?

File: frontend/src/components/header/Header.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  background: var(--primary-dark);
+  padding: 20px 30px 13px 20px;
+  border-radius: 0 0 50px 0;
+`
+
+const Heading = styled.h1`
+  color: var(--primary-light);
+  font-size: 1.75rem;
+  font-weight: 700;
+  line-height: 1;
```
Comment (felix-horn) [inline frontend/src/components/header/Header.js:27]: Wouldn't that be s.th. for the GlobalStlye.js?

File: frontend/src/components/header/Header.test.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { render } from '@testing-library/react'
+import 'jest-styled-components'
+import Header from './Header'
+
+describe('Header', () => {
+  it('renders correctly', () => {
+    const { container, getByText } = render(<Header />)
+
+    expect(container.firstChild).toMatchSnapshot()
+    expect(getByText(/My Plants/i)).toBeInTheDocument()
```
Comment (felix-horn) [inline frontend/src/components/header/Header.test.js:10]: In this case you could also just write `...getByText("My Plants")). ...`

File: frontend/src/components/header/Header.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+const HeaderStyled = styled.header`
+  display: flex;
+`
+
+const HeadingContainer = styled.div`
+  background: var(--primary-dark);
+  padding: 20px 30px 13px 20px;
+  border-radius: 0 0 50px 0;
+`
```
Comment (felix-horn) [inline frontend/src/components/header/Header.js:21]: I like how you defined each element individually and didn't nest anything! 👍 Very clean!

File: frontend/src/data/plants.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    "species": "Monstera deliciosa"
+  },
+  {
+    "id": 2,
+    "name": "Birgitt",
+    "species": "Monstera adansonii"
+  },
+  {
+    "id": 3,
+    "name": "Karen",
```
Comment (felix-horn) [inline frontend/src/data/plants.json:14]: "She's being a total Karen..."

File: frontend/src/components/plantcard/PlantCard.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  justify-items: center;
+`
+
+const ImgContainer = styled.div`
+  background-image: url(${(props) => props.src});
+  background-position: center;
+  background-size: cover, contain;
+  border-radius: 50%;
+  height: 100px;
+  margin-bottom: 10px;
```
Comment (felix-horn) [inline frontend/src/components/plantcard/PlantCard.js:31]: I try to use margins as seldom as possible and rather try to manipulate the distance of items with their parent. E.g. grid / gap. But I'm not too sure whether that's an option here.

File: frontend/src/components/plantcard/PlantCard.test.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  it('renders correctly', () => {
+    const { container, getByText } = render(
+      <PlantCard name={'Jack'} species={'Monstera deliciosa'} />
+    )
+
+    expect(container.firstChild).toMatchSnapshot()
+    expect(getByText(/Jack/i)).toBeInTheDocument()
+    expect(getByText(/Monstera deliciosa/i)).toBeInTheDocument()
+  })
+})
```
Comment (felix-horn) [inline frontend/src/components/plantcard/PlantCard.test.js:15]: Uuuuhhh... Testy McTestinator :)) 👍

File: frontend/src/plantlistpage/PlantListPage.test.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+    expect(container.firstChild).toMatchSnapshot()
+  })
+
+  it('displays the plant names and species', () => {
+    const { getByText } = render(<PlantListPage />)
+
+    plants.forEach(({ name }) => expect(getByText(name)).toBeInTheDocument())
+    plants.forEach(({ species }) =>
+      expect(getByText(species)).toBeInTheDocument()
```
Comment (felix-horn) [inline frontend/src/plantlistpage/PlantListPage.test.js:18]: Cooool! I haven't seen that one before!! I think I will copy it :)

**Review by felix-horn — COMMENTED** [review]
Very clean and straightforward! I didn't expect anything less from you :) Cool 👍

File: frontend/src/components/plantcard/PlantCard.js (hunk 6 above)
Comment (FabianAlleblas) [reply frontend/src/components/plantcard/PlantCard.js:31]: grid gap would affect the two paragraphs, I could wrap them, but them I have to position them again.

File: frontend/src/plantlistpage/PlantListPage.test.js (hunk 8 above)
Comment (FabianAlleblas) [reply frontend/src/plantlistpage/PlantListPage.test.js:18]: Copied it from the Student Companion App lel

File: frontend/src/App.js (hunk 1 above)
Comment (FabianAlleblas) [reply frontend/src/App.js:6]: Gonna test it 👍

File: frontend/src/components/plantcard/PlantCard.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  width: 100px;
+`
+
+const PlantName = styled.p`
```
Comment (ninapeters) [inline frontend/src/components/plantcard/PlantCard.js:35]: Semantically, I don't think `p` is quite right here. Maybe an `h` or a `span` fits better? For `PlantSpecies` it's the same.

File: frontend/src/data/plants.json [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    "species": "Rhaphidophora tetrasperma"
+  },
+  {
+    "id": 5,
+    "name": "Samuel",
+    "species": "Alocasia zebrina"
+  },
+  {
+    "id": 6,
+    "name": "Greg",
```
Comment (ninapeters) [inline frontend/src/data/plants.json:29]: 🐸

File: frontend/src/globalstyles/GlobalStyle.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    --warning-color: rgb(229,27,27);
+    --bar-bg-color: rgb(229,229,229);
+    --bar-water-color: rgb(27,108,229);
+    --bar-fertilizer-color: rgb(225,199,89);
+}
+
+* {
+    box-sizing: border-box;
+    padding: 0;
+    margin: 0;
```
Comment (ninapeters) [inline frontend/src/globalstyles/GlobalStyle.js:20]: I would define margin and padding in the body.

**Review by ninapeters — COMMENTED** [review]
Awesome! I look forward to the next steps in your app.

File: frontend/src/components/plantcard/PlantCard.js (hunk 9 above)
Comment (FabianAlleblas) [reply frontend/src/components/plantcard/PlantCard.js]: h5 and h6 it is... merci

**Pull request merged**